_articles/front-proxy.md

```diff
@@ -0,0 +1,227 @@
+---
+
+[//]: # ( Copyright 2018 Turbine Labs, Inc.                                   )
+[//]: # ( you may not use this file except in compliance with the License.    )
+[//]: # ( You may obtain a copy of the License at                             )
+[//]: # (                                                                     )
+[//]: # (     http://www.apache.org/licenses/LICENSE-2.0                      )
+[//]: # (                                                                     )
+[//]: # ( Unless required by applicable law or agreed to in writing, software )
+[//]: # ( distributed under the License is distributed on an "AS IS" BASIS,   )
+[//]: # ( WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or     )
+[//]: # ( implied. See the License for the specific language governing        )
+[//]: # ( permissions and limitations under the License.                      )
+
+[//]: # (Front Proxy)
+
+## Front Proxy Deployment
+
+One of the most powerful ways to deploy Envoy is as a front proxy. In this
+configuration, Envoy acts as the primary load balancer for customer requests
+from the public internet. Managing this public traffic—also called north-south
+traffic—is something that every web app must do, so it’s a natural place to
+start with Envoy. This kind of deployment style is also called an edge proxy or
+simply a load balancer.
+
+Starting with a front proxy is straightforward, because the first step is
+simply to port whatever existing configuration you have to Envoy. When you
+deploy your Envoy-based front proxy, you can set it up in parallel to your
+current setup, giving you the change to test and evaluate Envoy before sending
+production traffic through it. Once Envoy is stable and tested, it provides a
+high-leverage place to start enabling Envoy’s more advanced features around
+resilience and observability for your whole organization.
+
+## Deploying a Front Proxy
+
+Though Envoy is capable enough to be deployed right at the edge of your
+network, most public cloud providers expose layer 3/4 load balancers with more
+capabilities than you need. The typical deployment will use
+[AWS’ NLB](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html),
+[GCP’s Regional NLB](https://cloud.google.com/compute/docs/load-balancing/network/),
+or a minimal configuration of a more feature-rich load balancer
+like AWS ALB. These baked-in load balancers can handle regional or global
+traffic balancing, which is impossible for a set of VMs running Envoy (which
+are necessarily constrained to a single availability zone) to handle.
+
+A simple, robust deployment of an Envoy front proxy uses an autoscaling group
+of Envoys based on network traffic. Based on where your routing rules currently
+live (NGINX config files, AWS ALB configuration, etc.), you will need to port
+these to a set of Envoy routes. See
+[Routing Configuration](https://www.learnenvoy.io/articles/routing-configuration.html) for more details.
```

Comment:
wrap

Reply:
also would it make more sense to do relative links for internal links? not sure it matters, though we do it below for SSL / retries / health checks

Reply:
yea I can do relatives